R/modules.R

```diff
@@ -31,6 +31,14 @@ createSessionProxy <- function(parentSession, ...) {
   # but not `session$userData <- TRUE`) from within a module
   # without any hacks (see PR #1732)
   if (identical(x[[name]], value)) return(x)
+
+  # Special case for $options (issue #3112)
+  if (name == "options") {
+    parent <- .subset2(x, "parent")
```

Comment:
This this work for a module inside a module? (2x parents up the tree)

Reply:
I updated to work with nested modules. I've also updated the test app.